src/templates/healthcert/vaccinationCertTemplate.tsx

```diff
@@ -45,40 +59,117 @@ export const VaccinationCertTemplate: FunctionComponent<TemplateProps<NotarisedH
   ) as ImmunizationRecommendation;
 
   const passportNumber = document.notarisationMetadata?.passportNumber;
-  const patientName = typeof patient?.name?.[0] === "object" ? patient.name[0].text : "";
+  const patientName = typeof patient?.name?.[0] === "object" ? patient?.name[0]?.text : "";
   const patientNric = patient?.identifier?.find(isNric)?.value || "";
   const patientNationalityCode =
     patient?.extension?.find(
       extension => extension.url === "http://hl7.org/fhir/StructureDefinition/patient-nationality"
     )?.code.text || "";
-  const patientBirthDate = formatDate(patient.birthDate || "");
+  const patientBirthDate = formatDate(patient?.birthDate || "");
   const effectiveDate = formatDate(recommendation?.recommendation[0].dateCriterion[0].value);
 
   const url = (document.notarisationMetadata as any)?.url;
+  const encryptedEuHealthCert = (document.notarisationMetadata as any)?.encryptedEuHealthCert;
   const memoSections: JSX.Element[] = [];
 
-  memoSections.push(
-    <VaccinationMemoSection
-      immunizations={immunizations.map(simplifyImmunizationObjectWithLocation(locations))}
-      effectiveDate={effectiveDate}
-      patientName={patientName}
-      patientNric={patientNric}
-      patientNationalityCode={patientNationalityCode}
-      patientBirthDate={patientBirthDate}
-      passportNumber={passportNumber}
-    />
-  );
+  const mappedImmunizations = immunizations.map(simplifyImmunizationObjectWithLocation(locations));
+  if (encryptedEuHealthCert) {
+    memoSections.push(
+      <VaccinationV2MemoSection
+        immunizations={mappedImmunizations}
+        patientName={patientName}
+        patientNric={patientNric}
+        patientNationalityCode={patientNationalityCode}
+        patientBirthDate={patientBirthDate}
+        passportNumber={passportNumber}
+      />
+    );
 
-  return (
-    <Page className={className}>
-      <Background />
-      <Logo src={document.logo} alt="healthcare provider logo" />
-      {memoSections}
-      {url && (
-        <QrCodeContainer>
-          <QRCode value={url} level={"M"} size={200} />
-        </QrCodeContainer>
-      )}
-    </Page>
-  );
+    return (
+      <>
+        <Page className={className}>
+          <Background />
+          <Logo src={document.logo} alt="healthcare provider logo" />
+          {memoSections}
+        </Page>
+        <Page className={className}>
+          <ResultSection>
+            <p>
+              {patientName} has been vaccinated with {mappedImmunizations[0]?.vaccineName} effective from{" "}
+              {effectiveDate}.
+            </p>
+          </ResultSection>
+          <TravellerInfoSection>
+            Note: Travellers are subject to the country or region&apos;s requirements prior to travel.
+            <br />
+            The QR code used for verification is based on your <u>destination country</u>.
+          </TravellerInfoSection>
+          <br />
+          {encryptedEuHealthCert && (
```

Comment:
encryptedEuHealthCert should always be defined here since the `if` clause above covered it